Add User Signup/Login/Logout.

mostlybadfly opened the pull request:
Added a signup and login form along with the contrib.auth views. Currently it takes a username and password. Email signup/confirmation/reset might be implemented later. Still working on a profile view and will add to this branch.

mostlybadfly committed:
Add User Signup/Login/Logout.

skain commented on recipes/models.py:
```diff
@@ -18,3 +20,12 @@ class Instruction(models.Model):
     instruction_text = models.TextField(null=True)
     recipe = models.ForeignKey(Recipe, related_name='instructions')
+
+class Profile(models.Model):
```
I don't understand the point of the Profile class. Did you copy that from somewhere or come up with it yourself?

mostlybadfly replied:
It was from the sign up guide, but I think I might have gotten confused on the usage, because it doesn’t really look like it is doing anything. I can probably remove that part

skain reviewed (approved):
Other than my confusion about the Profile class this all looks OK I think. As to where user stuff goes, you have two options I believe: 1) You do it like you've done it and just put it in the recipes app (you could make a templates/user folder for user-specific templates and a users package within recipes if organization becomes a problem) or 2) you could create a separate User or Authentication django app within your project. 

For a relatively simple app like this I think option 1 is fine.

mostlybadfly committed:
Removed Profile from model. Moved signup to registration templates.

mostlybadfly commented:
removed the Profile stuff as mentioned. `User` is signing up , logging in and out. I added `signup.html` to `templates/registration/` since django auth views uses this for custom templates. Going to merge this to work on some other views.
